src/components/Chat/InputMessageV2.tsx

```diff
@@ -119,19 +138,187 @@ const InputMessageV2 = ({
+        url: videoPath,
+        timeStamp: 1000
+      });
+      return thumbnail.path;
+    } catch (error) {
+      console.error('Error generating thumbnail', error);
+      return null;
+    }
+  };
+
+  const processMediasSequentially = async (medias, delayTime) => {
+    const newMedias: any = [];
+    for (const media of medias) {
+      if (media?.mime?.includes('video')) {
+        const thumbnailPath = await getVideoThumbnail(media.path);
+        newMedias.push({type: 'video', path: thumbnailPath, pathVideo: media.path});
+      } else {
+        const croppedImage = await processImage(media);
+        newMedias.push({type: 'photo', path: croppedImage});
+        await delay(delayTime);
+      }
+    }
+    return newMedias;
+  };
+
+  const onOpenMedia = async () => {
+    const {success} = await requestExternalStoragePermission();
+    if (success) {
+      ImagePicker.openPicker({
+        multiple: true,
+        maxFiles: 20
+      }).then(async (medias) => {
+        const newMedias = await processMediasSequentially(medias, 200);
+
+        handleUploadMedia(newMedias);
+      });
+    } else {
+      openAlertPermission(
+        'We’re not able to access your photos, please adjust your permission settings for BetterSocial.'
+      );
+    }
+  };
+
+  const onOpenCamera = async () => {
+    const {success} = await requestCameraPermission();
+    if (success) {
+      ImagePicker.openCamera({
+        mediaType: 'photo'
+      }).then(async (imageRes) => {
+        const imageCropped = await ImagePicker.openCropper({
+          mediaType: 'photo',
+          path: imageRes.path,
+          cropperChooseText: 'Next',
+          freeStyleCropEnabled: true
+        });
+        handleUploadCamera(imageCropped.path);
+      });
+    } else {
+      openAlertPermission(
+        'We’re not able to access your camera, please adjust your permission settings for BetterSocial.'
+      );
+    }
+  };
+
+  const onOpenFile = async () => {
+    const {success} = await requestExternalStoragePermission();
+    if (success) {
+      DocumentPicker.pickSingle({
+        presentationStyle: 'fullScreen',
+        copyTo: 'cachesDirectory'
+      }).then((pickerResult) => {
+        console.warn('pickerResult', pickerResult);
+        //
+      });
+    } else {
+      openAlertPermission(
+        'We’re not able to access your document library, please adjust your permission settings for BetterSocial.'
+      );
+    }
+  };
 
   const onChangeInput = (v) => {
     setText(v);
   };
-  const onSelectEmoji = () => {
-    refEmoji.current.close();
+
+  const onOpenGIF = () => {
+    refAttachment.current.close();
+    setTimeout(() => {
+      refGif.current.open();
+    }, 500);
+  };
+
+  const onCloseGIF = () => {
+    refGif.current.close();
+  };
+
+  const onSelectAttachment = () => {
+    refAttachment.current.open();
   };
 
   const handleSendMessage = () => {
-    onSendButtonClicked(text);
+    onSendButtonClicked(text, []);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This large block of code introduces several new functions and state variables related to handling attachments in chat messages. While the logic seems sound, it might be beneficial to break down some of these larger functions into smaller, more manageable pieces. This would improve readability and make it easier to test individual pieces of functionality.

For example, the `handleUploadMedia` and `handleUploadCamera` functions could be broken down into smaller functions that handle each step of the process (e.g., uploading the media, creating the result URLs, sending the message).



<!-- This is an auto-generated comment by OSS CodeRabbit -->